src/iota/ed25519.c

```diff
@@ -22,40 +22,39 @@ uint8_t ed25519_get_key_pair(uint32_t *bip32_path, uint32_t bip32_path_length,
                              cx_ecfp_private_key_t *pk,
                              cx_ecfp_public_key_t *pub)
 {
-    uint8_t keySeed[32];
-    uint8_t ret = 1;
+    uint8_t keySeed[64];
 
     // getting the seed to derive and configuring it with SLIP10
-    os_perso_derive_node_bip32_seed_key(
-        HDW_ED25519_SLIP10, CX_CURVE_Ed25519, bip32_path, bip32_path_length,
-        keySeed, NULL, (unsigned char *)"ed25519 seed", 12);
-
-    BEGIN_TRY
-    {
-        TRY
-        {
-            // initializing the private key and public key instance
-            // with selected curve ED25519
-            cx_ecfp_init_private_key(CX_CURVE_Ed25519, keySeed, sizeof(keySeed),
-                                     pk);
-            cx_ecfp_init_public_key(CX_CURVE_Ed25519, NULL, 0, pub);
-
-            // generating the key pair
-            cx_ecfp_generate_pair(CX_CURVE_Ed25519, pub, pk, 1);
+    cx_err_t err = CX_OK;
+    do {
+        err = os_derive_bip32_with_seed_no_throw(
```

Comment:
@xchapron-ledger this is a good suggestion. I tried to include the `ledger_standard_app` and run into a couple of problems. First thing was a name-clash with my `macros.h` what would be no issue to solve. Another thing is that it comes with a `main` and althought it would be a good idea to use the "official" framework I just can't do it because I don't have time to absolutely test everything from scratch because it changes how exceptions are handled as well. Not possible currently. So I will copy the two functions and replace my own two with your variant :see_no_evil:

Reply:
@shufps I would recommand only including crypto_helpers.c to the build and using it from SDK lib_standard_app. Here is an example:
- https://github.com/LedgerHQ/app-bitcoin-new/pull/167/commits/54b2831c6cb5f724aedab952dca5da59606d8097 (the INCLUDE_PATH part is no longueur needed).
- https://github.com/LedgerHQ/app-bitcoin-new/pull/167/commits/daa18285edaa8815d4fd19f4d3ea4f1a6769af19

Reply:
I changed it like you suggested :raised_hands:

Reply:
Nice plus is that the changes I had to do to fix deprecation warnings actually vanished again and everything it got easier

Reply:
btw, it seems your "reusable workflows" are broken because the `latest` tag of the `ledger-app-builder-lite` doesn't exist anymore.